src/machine/machine_nrf.go

```diff
@@ -258,12 +263,19 @@ func (i2c I2C) Configure(config I2CConfig) {
 // Tx does a single I2C transaction at the specified address.
 // It clocks out the given address, writes the bytes in w, reads back len(r)
 // bytes and stores them in r, and generates a stop condition on the bus.
-func (i2c I2C) Tx(addr uint16, w, r []byte) error {
+func (i2c I2C) Tx(addr uint16, w, r []byte) (err error) {
 	i2c.Bus.ADDRESS.Set(uint32(addr))
+	defer func() {
```

Comment:
`defer` is best avoided especially in such low level code as it can be hard to optimize. For example, I have been looking into `recover` support and `recover` support will make `defer` much more expensive.

I would suggest using a `goto` here for error handling, for details: https://eli.thegreenplace.net/2009/04/27/using-goto-for-error-handling-in-c

Reply:
Yes, I'm familiar with this style, like in the Linux source code. I think it's not idiomatic Go, but I agree with you that microcontrollers require special considerations.

I'm modifying it all.  Thanks!  :-)